From 85e406b23c086c446ccfd6f8c83670800cc4de3a Mon Sep 17 00:00:00 2001
From: Patrick Roy <roypat@amazon.co.uk>
Date: Tue, 30 Jul 2024 13:59:04 +0100
Subject: [PATCH] fix: Adjust config options for what is needed by our tests

Additionally, set `CONFIG_MSDOS_PARTITION=y` for our PARTUUID tests,
`CONFIG_DEVMEM=y` on aarch64/5.10 (for various integration tests that
rely on /dev/mem), and CONFIG_IKCONFIG=y and CONFIG_IKCONFIG_PROC=y for
the spectre-meltdown-checker.sh tests to work on AMD.

Signed-off-by: Patrick Roy <roypat@amazon.co.uk>
---
 .../guest_configs/microvm-kernel-ci-aarch64-4.14.config    | 5 +++--
 .../guest_configs/microvm-kernel-ci-aarch64-5.10.config    | 7 ++++---
 .../guest_configs/microvm-kernel-ci-x86_64-4.14.config     | 5 +++--
 .../microvm-kernel-ci-x86_64-5.10-no-acpi.config           | 5 +++--
 .../guest_configs/microvm-kernel-ci-x86_64-5.10.config     | 5 +++--
 5 files changed, 16 insertions(+), 11 deletions(-)

diff --git a/resources/guest_configs/microvm-kernel-ci-aarch64-4.14.config b/resources/guest_configs/microvm-kernel-ci-aarch64-4.14.config
index 694058ca..350f0f48 100644
--- a/resources/guest_configs/microvm-kernel-ci-aarch64-4.14.config
+++ b/resources/guest_configs/microvm-kernel-ci-aarch64-4.14.config
@@ -118,7 +118,8 @@ CONFIG_TREE_SRCU=y
 CONFIG_RCU_STALL_COMMON=y
 CONFIG_RCU_NEED_SEGCBLIST=y
 # CONFIG_BUILD_BIN2C is not set
-# CONFIG_IKCONFIG is not set
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
 CONFIG_LOG_BUF_SHIFT=21
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=12
 CONFIG_PRINTK_SAFE_LOG_BUF_SHIFT=13
@@ -344,7 +345,7 @@ CONFIG_PARTITION_ADVANCED=y
 # CONFIG_AMIGA_PARTITION is not set
 # CONFIG_ATARI_PARTITION is not set
 # CONFIG_MAC_PARTITION is not set
-# CONFIG_MSDOS_PARTITION is not set
+CONFIG_MSDOS_PARTITION=y
 # CONFIG_LDM_PARTITION is not set
 # CONFIG_SGI_PARTITION is not set
 # CONFIG_ULTRIX_PARTITION is not set
diff --git a/resources/guest_configs/microvm-kernel-ci-aarch64-5.10.config b/resources/guest_configs/microvm-kernel-ci-aarch64-5.10.config
index ac44904c..0555055c 100644
--- a/resources/guest_configs/microvm-kernel-ci-aarch64-5.10.config
+++ b/resources/guest_configs/microvm-kernel-ci-aarch64-5.10.config
@@ -110,7 +110,8 @@ CONFIG_RCU_STALL_COMMON=y
 CONFIG_RCU_NEED_SEGCBLIST=y
 # end of RCU Subsystem
 
-# CONFIG_IKCONFIG is not set
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
 # CONFIG_IKHEADERS is not set
 CONFIG_LOG_BUF_SHIFT=17
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=12
@@ -752,7 +753,7 @@ CONFIG_PARTITION_ADVANCED=y
 # CONFIG_AMIGA_PARTITION is not set
 # CONFIG_ATARI_PARTITION is not set
 # CONFIG_MAC_PARTITION is not set
-# CONFIG_MSDOS_PARTITION is not set
+CONFIG_MSDOS_PARTITION=y
 # CONFIG_LDM_PARTITION is not set
 # CONFIG_SGI_PARTITION is not set
 # CONFIG_ULTRIX_PARTITION is not set
@@ -1726,7 +1727,7 @@ CONFIG_HW_RANDOM_VIRTIO=y
 # CONFIG_HW_RANDOM_CCTRNG is not set
 # CONFIG_HW_RANDOM_XIPHERA is not set
 # CONFIG_HW_RANDOM_GRAVITON is not set
-# CONFIG_DEVMEM is not set
+CONFIG_DEVMEM=y
 # CONFIG_RAW_DRIVER is not set
 # CONFIG_TCG_TPM is not set
 # CONFIG_XILLYBUS is not set
diff --git a/resources/guest_configs/microvm-kernel-ci-x86_64-4.14.config b/resources/guest_configs/microvm-kernel-ci-x86_64-4.14.config
index ee6df5ff..b4e65c6a 100644
--- a/resources/guest_configs/microvm-kernel-ci-x86_64-4.14.config
+++ b/resources/guest_configs/microvm-kernel-ci-x86_64-4.14.config
@@ -135,7 +135,8 @@ CONFIG_TREE_SRCU=y
 CONFIG_RCU_STALL_COMMON=y
 CONFIG_RCU_NEED_SEGCBLIST=y
 CONFIG_BUILD_BIN2C=y
-# CONFIG_IKCONFIG is not set
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
 CONFIG_LOG_BUF_SHIFT=21
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=12
 CONFIG_PRINTK_SAFE_LOG_BUF_SHIFT=13
@@ -395,7 +396,7 @@ CONFIG_PARTITION_ADVANCED=y
 # CONFIG_AMIGA_PARTITION is not set
 # CONFIG_ATARI_PARTITION is not set
 # CONFIG_MAC_PARTITION is not set
-# CONFIG_MSDOS_PARTITION is not set
+CONFIG_MSDOS_PARTITION=y
 # CONFIG_LDM_PARTITION is not set
 # CONFIG_SGI_PARTITION is not set
 # CONFIG_ULTRIX_PARTITION is not set
diff --git a/resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config b/resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config
index b87fb3e4..a27c1f84 100644
--- a/resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config
+++ b/resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config
@@ -134,7 +134,8 @@ CONFIG_RCU_NEED_SEGCBLIST=y
 # end of RCU Subsystem
 
 CONFIG_BUILD_BIN2C=y
-# CONFIG_IKCONFIG is not set
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
 # CONFIG_IKHEADERS is not set
 CONFIG_LOG_BUF_SHIFT=17
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=12
@@ -702,7 +703,7 @@ CONFIG_PARTITION_ADVANCED=y
 # CONFIG_AMIGA_PARTITION is not set
 # CONFIG_ATARI_PARTITION is not set
 # CONFIG_MAC_PARTITION is not set
-# CONFIG_MSDOS_PARTITION is not set
+CONFIG_MSDOS_PARTITION=y
 # CONFIG_LDM_PARTITION is not set
 # CONFIG_SGI_PARTITION is not set
 # CONFIG_ULTRIX_PARTITION is not set
diff --git a/resources/guest_configs/microvm-kernel-ci-x86_64-5.10.config b/resources/guest_configs/microvm-kernel-ci-x86_64-5.10.config
index 09461c17..af9ec662 100644
--- a/resources/guest_configs/microvm-kernel-ci-x86_64-5.10.config
+++ b/resources/guest_configs/microvm-kernel-ci-x86_64-5.10.config
@@ -127,7 +127,8 @@ CONFIG_RCU_NEED_SEGCBLIST=y
 # end of RCU Subsystem
 
 CONFIG_BUILD_BIN2C=y
-# CONFIG_IKCONFIG is not set
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
 # CONFIG_IKHEADERS is not set
 CONFIG_LOG_BUF_SHIFT=17
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=12
@@ -739,7 +740,7 @@ CONFIG_PARTITION_ADVANCED=y
 # CONFIG_AMIGA_PARTITION is not set
 # CONFIG_ATARI_PARTITION is not set
 # CONFIG_MAC_PARTITION is not set
-# CONFIG_MSDOS_PARTITION is not set
+CONFIG_MSDOS_PARTITION=y
 # CONFIG_LDM_PARTITION is not set
 # CONFIG_SGI_PARTITION is not set
 # CONFIG_ULTRIX_PARTITION is not set
-- 
2.34.1

